Testreglar/4.1.2/App/app-4.1.2a.json

```diff
@@ -1,128 +1,237 @@
-            "kilde": [],
-            "svarArray": [
-                "Knappar",
-                "Avkryssingsboks",
-                "Lenka bilder",
-                "Nedtrekksliste",
-                "Inputfelt",
-                "Anna"
-            ],
-            "ruting": {
-                "alle": {
-                    "type": "gaaTil",
-                    "steg": "3.3"
-                }
-            }
-        },
-        {
-            "stegnr": "3.3",
-            "spm": "Aktiver skjermlesar (opplesingsfunksjon).",
-            "ht": "<p>Trykk på det interaktive elementet for å få lest opp ledeteksten/den alternative teksten til elementet.</p>",
-            "type": "instruksjon",
-            "kilde": [],
-            "ruting": {
-                "alle": {
-                    "type": "gaaTil",
-                    "steg": "3.4"
-                }
-            }
-        },
-        {
-            "stegnr": "3.4",
-            "spm": "Får det interaktive elementet du testar opplese ein ledetekst som identifiserer elementet?",
-            "ht": "<p>Sjekk at teksten du får opplest for:</p>\n<ul>\n<li>skjemaelement identifiserer skjemaelementet</li>\n<li>lenka bilder beskriv lenkemålet</li>\n<li>knappar identifiserer funksjonen til knappen.</li>\n</ul>",
-            "type": "jaNei",
-            "kilde": [
-                "G82",
-                "F89",
-                "G135"
-            ],
-            "ruting": {
-                "ja": {
-                    "type": "avslutt",
-                    "fasit": "Ja",
-                    "utfall": "Det interaktive elementet får lest opp ein ledetekst som identifiserer elementet."
-                },
-                "nei": {
-                    "type": "avslutt",
-                    "fasit": "Nei",
-                    "utfall": "Det interaktive elementet får ikkje lest opp ein ledetekst som identifiserer elementet."
-                }
-            }
-        }
-    ]
+	"namn": "App-4.1.2a For brukergrensesnittkomponenter kan tilgjengelig navn, rolle og tilstand bestemmes programmatisk 2023",
+	"id": "app-4.1.2a",
+	"testlabId": 445,
+	"versjon": "1.0",
+	"type": "App",
+	"spraak": "nb",
+	"kravTilSamsvar": "<p>Det er flere måter å oppfylle kravet på.</p>\r\n<p>Navn og rolle:</p>\r\n<ul>\r\n<li>Alle brukergrensesnittkomponenter har et tilgjengelig navn, som beskriver formålet med den aktuelle komponenten og</li>\r\n<li>Brukergrensesnittkomponenter som tilhører en gruppe, er også koblet til et tilgjengelig navn som gjelder for gruppen og</li>\r\n<li>Alle brukergrensesnittkomponenter har riktig rolle, som identifiserer funksjonen til den aktuelle komponenten</li>\r\n</ul>\r\n<p>Tilstander, egenskaper og verdier:</p>\r\n<ul>\r\n<li>Når tilstander, egenskaper og verdier i brukergrensesnittkomponenter kan angis av brukeren, skal denne informasjonen også angis programmatisk og</li>\r\n<li>Varsel om endringer i den aktuelle komponenten er tilgjengelig for brukeragenter </li>\r\n</ul>",
```

Comment:
Er det flere måter å oppfylle kravene på, eller gjelder alle punktene for alle brukergrensesnittkomponenter.

Reply:
Alle kravene skal være oppfylt, men kulepunkt tre gjelder kun for gruppeelement. Det er flere måter å kode tilgjengelig navn på, så derfor står det at det er flere måter å oppfylle kravet på.

Reply:
Skal fjernes her og i andre publiserte testregler etter kommende statusmøte.